pkg/runcfanotify/runcfanotify.go

```diff
@@ -413,16 +413,19 @@ func (n *RuncNotifier) monitorRuncInstance(bundleDir string, pidFile string) err
 
 	// watchPidFileIterate() will read config.json and it might be in the
 	// same directory as the pid file. To avoid getting events unrelated to
-	// the pidfile, add an ignore mask.
+	// the pidfile, add an ignore mask. Same for passwd in case non-root
+	// crio containers
 	//
 	// This is best effort because the ignore mask is unfortunately not
 	// respected until a fix in Linux 5.9:
 	// https://github.com/torvalds/linux/commit/497b0c5a7c0688c1b100a9c2e267337f677c198e
-	configJSONPath := filepath.Join(bundleDir, "config.json")
-	err = pidFileDirNotify.Mark(unix.FAN_MARK_ADD|unix.FAN_MARK_IGNORED_MASK, unix.FAN_ACCESS_PERM, unix.AT_FDCWD, configJSONPath)
-	if err != nil {
-		pidFileDirNotify.File.Close()
-		return fmt.Errorf("ignoring %s: %w", configJSONPath, err)
+	ignoreFiles := []string{"config.json", "passwd"}
+	for _, ignoreFile := range ignoreFiles {
+		err = pidFileDirNotify.Mark(unix.FAN_MARK_ADD|unix.FAN_MARK_IGNORED_MASK, unix.FAN_ACCESS_PERM, unix.AT_FDCWD, filepath.Join(pidFileDir, ignoreFile))
+		if err != nil {
+			pidFileDirNotify.File.Close()
+			return fmt.Errorf("ignoring %s: %w", ignoreFiles, err)
```

Comment:
What if the file `passwd` does not exist? Should we ignore the ENOENT error in that case?

Reply:
Good point. Yes, we should. I see the CI failing because of this as well. I will update it.

Reply:
Done. I was wondering if we should have a fallback check in `watchPidFileIterate` to handle this e.g `filepath.Base(path) == filepath.Base(pidFileDir)` not sure how robust that is!